include/rocksdb/env.h

```diff
@@ -769,6 +777,12 @@ class WritableFile {
   // PositionedAppend, so the users cannot mix the two.
   virtual Status Append(const Slice& data) = 0;
 
+  // Append data with verification information
+  virtual Status AppendWithVerify(
```

Comment:
Is there a need to define these new interfaces in ```WritableFile```? Since we want to deprecate the storage part of ```Env``` eventually, why not just define them in ```file_system.h``` and redirect in ```LegacyWritableFileWrapper```. Users like Zippy can work with the ```FileSystem``` directly or call ```env->GetFileSystem()->AppendWithVerify()```.